utilities/visual_regression.js

```diff
@@ -110,17 +108,23 @@ async function compareScreenshots() {
       screenshot
     );
 
-    // Diff the images asynchronously.
-    const diffResult = await fromNode((cb) => {
-      imageDiff.getFullResult({
-        actualImage: sessionImagePath,
-        expectedImage: baselineImagePath,
-        diffImage: diffImagePath,
-        shadow: true,
-      }, cb);
-    });
+    const sessionImage = PNG.sync.read(await fs.promises.readFile(sessionImagePath));
+    const baselineImage = PNG.sync.read(await fs.promises.readFile(baselineImagePath));
+    const { width, height } = sessionImage;
+    const diff = new PNG({ width, height });
+
+    const numDiffPixels = pixelmatch(
```

Comment:
@cjcenizal I've replaced the `image-diff` module with `pixelmatch` as `image-diff` is deprecated and makes use of deprecated API's in Node core. However, I've not been able to verify if this new code works. As far as I can see this is used to create a "diff gallery" in HTML form. What's the best way for me to run this and see the output?

Reply:
@elastic/kibana-qa Just realized that CJ is on leave currently. Do any of you know about this part of the code?

Reply:
FYI, I'm looking at test `dashboard_app_using_current_data_dashboard_snapshots_compare_TSVB_snapshot/`
in the Chrome UI Functional Tests

This looks correct (it's headless Chome so this is the simplest case);
```
[00:16:09]                 │ debg ======browser======== setWindowSize 1000 500
[00:16:09]                 │ debg ======browser======== actual initial screenshot size width=1200, height=800
[00:16:10]                 │ debg ======browser======== actual second screenshot size width= 600, height=400
[00:16:10]                 │ debg ======browser======== calculated values xBorder= 0, yBorder=0, xScaling=1, yScaling=1
[00:16:10]                 │ debg ======browser======== setting browser size to 1000 x 500
[00:16:10]                 │ debg ======browser======== final screenshot size width=1000, height=500
```

And the actual screenshot and comparison works correctly;
```
[00:16:43]                 │ info Taking screenshot "/dev/shm/workspace/kibana/test/functional/screenshots/session/tsvb_dashboard.png"
[00:16:43]                 │ debg comparePngs: /dev/shm/workspace/kibana/test/functional/screenshots/session/tsvb_dashboard.png vs /dev/shm/workspace/kibana/test/functional/screenshots/baseline/tsvb_dashboard.png
[00:16:44]                 │ debg calculating diff pixels...
[00:16:44]                 │ debg percent different: 0.012874
```

Reply:
Phew! Glad that it worked - it's pretty rare that writing code without running it resolves in a functional program in the first go 😅Thanks for testing it.